src/components/settings/EnvironmentVariables/index.tsx

```diff
@@ -0,0 +1,270 @@
+import { useEffect } from 'react'
+import { useForm, FormProvider } from 'react-hook-form'
+import { Paper, Grid, Typography, TextField, Button, Tooltip, IconButton, SvgIcon } from '@mui/material'
+import InputAdornment from '@mui/material/InputAdornment'
+import RotateLeftIcon from '@mui/icons-material/RotateLeft'
+import { useAppDispatch, useAppSelector } from '@/store'
+import { selectSettings, setCgw, setRpc, setTenderly } from '@/store/settingsSlice'
+import { GATEWAY_URL_PRODUCTION, TENDERLY_SIMULATE_ENDPOINT_URL } from '@/config/constants'
+import useChainId from '@/hooks/useChainId'
+import { useCurrentChain } from '@/hooks/useChains'
+import { SETTINGS_EVENTS, trackEvent } from '@/services/analytics'
+import InfoIcon from '@/public/images/notifications/info.svg'
+import ExternalLink from '@/components/common/ExternalLink'
+
+export enum EnvVariablesField {
+  cgw = 'cgw',
+  rpc = 'rpc',
+  tenderlyURL = 'tenderlyURL',
+  tenderlyToken = 'tenderlyToken',
+}
+
+export type EnvVariablesFormData = {
+  [EnvVariablesField.cgw]: string
+  [EnvVariablesField.rpc]: string
+  [EnvVariablesField.tenderlyURL]: string
+  [EnvVariablesField.tenderlyToken]: string
+}
+
+const EnvironmentVariables = () => {
+  const chainId = useChainId()
+  const chain = useCurrentChain()
+  const settings = useAppSelector(selectSettings)
+  const dispatch = useAppDispatch()
+
+  const formMethods = useForm<EnvVariablesFormData>({
+    mode: 'onChange',
+    defaultValues: {
+      [EnvVariablesField.cgw]: settings.env.cgw ?? '',
+      [EnvVariablesField.rpc]: settings.env.rpc[chainId] ?? '',
+      [EnvVariablesField.tenderlyURL]: settings.env.tenderly.url ?? '',
+      [EnvVariablesField.tenderlyToken]: settings.env.tenderly.accessToken ?? '',
+    },
+  })
+
+  const { register, handleSubmit, reset, setValue, watch } = formMethods
+
+  const cgw = watch(EnvVariablesField.cgw)
+  const rpc = watch(EnvVariablesField.rpc)
+  const tenderlyURL = watch(EnvVariablesField.tenderlyURL)
+  const tenderlyToken = watch(EnvVariablesField.tenderlyToken)
+
+  const onSubmit = handleSubmit((data) => {
+    trackEvent({ ...SETTINGS_EVENTS.ENV_VARIABLES.SAVE })
+    dispatch(setCgw(data[EnvVariablesField.cgw]))
+    dispatch(setRpc({ chainId, rpc: data[EnvVariablesField.rpc] }))
+    dispatch(
+      setTenderly({ url: data[EnvVariablesField.tenderlyURL], accessToken: data[EnvVariablesField.tenderlyToken] }),
+    )
```

Comment:
Can we avoid dispatching 3 actions? 2 options I see: dispatch conditionally (if changes) or an action to dispatch the whole env object

Reply:
Makes sense as they are not being used anywhere else. I removed the individual setters and added a single `setEnv` action instead.